src/include/serial_inline.h

```diff
@@ -257,6 +257,12 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
 
     /* Mark the page dirty after updating the footprint. */
     __wt_page_modify_set(session, page);
+    /*
+     * We don't want to remove obsolete updates in the history store, since another reader might be
+     * reading these updates.
```

Comment:
Can you add a comment to explain the reason behind why an inserted update is possible to be globally visible and why the history store is accessed even when the data is globally visible? Otherwise, we will forget the reason behind the check.

Reply:
done! thanks hari